Gems/Atom/Feature/Common/Assets/Shaders/Materials/EnhancedPBR/EnhancedPBR_PixelGeometry.azsli

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// This #define magic lets you use the EvaluatePixelGeometry function in this file without making it the final EvaluatePixelGeometry
+// used in your shader. Simply #define EvaluatePixelGeometry to your custom definition before including this file
+//
+#ifndef EvaluatePixelGeometry
+#define EvaluatePixelGeometry(IN, isFrontFace)      EvaluatePixelGeometry_EnhancedPBR(IN, isFrontFace)
+#endif
+
+#include "../StandardPBR/StandardPBR_PixelGeometry.azsli"
+
+#include <Atom/Features/Debug.azsli>
+
+PixelGeometryData EvaluatePixelGeometry_EnhancedPBR(
+    inout float4 positionSV,
+    float3 positionWS,
+    float3 normal,
+    float3 tangent,
+    float3 bitangent,
+    float2 uvs[UvSetCount],
+    float2 detailUvs[UvSetCount],
+    bool isFrontFace)
+{
+    bool evaluateTangentFrame = ShouldHandleParallax() || o_normal_useTexture || (o_clearCoat_enabled && o_clearCoat_normal_useTexture) || o_detail_normal_useTexture;
```

Comment:
Nit: suggest caching the value of `ShouldHandleParallax`

Reply:
Thanks, excellent catch, will do